apps/oxfmt/src-js/index.ts

```diff
@@ -1,2 +1,113 @@
-export * from "./bindings.js";
-export { setupConfig, formatEmbeddedCode, formatFile } from "./prettier-proxy.js";
+import { format as napiFormat } from "./bindings.js";
+import { setupConfig, formatEmbeddedCode, formatFile } from "./prettier-proxy.js";
+
+export async function format(fileName: string, sourceText: string, options?: FormatOptions) {
+  if (typeof fileName !== "string") throw new TypeError("`fileName` must be a string");
+  if (typeof sourceText !== "string") throw new TypeError("`sourceText` must be a string");
+
+  return napiFormat(
+    fileName,
+    sourceText,
+    options ?? {},
+    setupConfig,
+    formatEmbeddedCode,
+    formatFile,
```

Comment:
Why do we need to pass the `formatFile` function to the Rust side? Can we determine if the file should delegate to `Prettier` on the JS side? If so, that will be more performant and avoid some mutual calling overhead.